model_evaluation/rfc.md

```diff
@@ -0,0 +1,91 @@
+|     mode                   |   String        |   Options: [score, predict, compute_metrics]  Required: True  Description: Model Evaluation mode ->         1. score -> Do prediction and compute metrics for predictions        2. predict -> Compute predictions only over given dataset        3. compute_metrics -> Compute metrics over predictions provided by user                                                                                                                                                                         |
+|     task                   |   String        |   Options: [classification, regression, forecasting]  Required: True  Description: Task type for which the model is trained                                                                                                                                                                                                                                                                                                                                                                       |
+|     model_uri              |   String        |   Required: False (Required only if mode is score or predict)  Description: MLFlow model uri (could be of type) ->        To Fetch models from an azureml run        - runs:/<azureml_run_id>/run-relative/path/to/model        To Fetch model from azureml model registry        - models:/<model_name>/<model_version>        - models:/<model_name>/<stage>  NOTE: model_uri has been added to accommodate current ML Designer as it doesn’t allow users to drag-and-drop Registered models    |
+|     mlflow_model           |   MLFlow Model  |   Required: False (Required only if mode is score or predict)  Description: MLFlow model (either registered or output of another job)                                                                                                                                                                                                                                                                                                                                                             |
+|     test_data              |   MLTable       |   Required: True  Description: Test Dataset for model evaluation. Allowed formats: Delimited, JSON Lines, Parquet and Plain Text (for some tasks).                                                                                                                                                                                                                        |
+|     evaluation_config      |   MLTable       |   Required: False  Description: Additional config files required by model/metrics package. This data asset should contain a JSON Config file and any number artifacts required to load/run the model. The MLTable file should only mention JSON file name in path.        Contents of JSON File:        - See Additional Parameters list below                               |
```

Comment:
In the previous proposal https://github.com/Azure/azureml-assets/pull/168#discussion_r953316621 the parameters were provided as a `String`, which was a UX pain point (see other comment).

But if we ask the users to provide their parameters as a JSON file it might be more painful even. If we take the scenario of a user of Designer UI, how do they even create this JSON file? They'd need to first go and upload it somewhere before they can play with the component.

Also, to change those parameters, the users will have to re-upload, then go into the UI and change the input to point to the new uploaded location.

Reply:
At the beginning in our first draft of component we had all of these additional parameters as part of the inputs of component directly (instead of JSON file). But, from an internal architecture review we got a feedback that its better user provides them as a dictionary or json file because the list was too long. 

Going ahead, we have a plan of moving away from command component and creating a custom component with custom UI. With custom component we'll be able to handle it in a better way and we'll have the custom validations as well.